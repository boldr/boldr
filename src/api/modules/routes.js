import express from 'express';

import authRoutes from './auth/auth.routes';
import collectionRoutes from './collection/collection.routes';
import galleryRoutes from './gallery/gallery.routes';
import linkRoutes from './link/link.routes';
import mediaRoutes from './media/media.routes';
import navigationRoutes from './navigation/navigation.routes';
import pageRoutes from './page/page.routes';
import postRoutes from './post/post.routes';
import roleRoutes from './role/role.routes';
import s3Router from './s3/s3.routes';
import settingRoutes from './setting/setting.routes';
import tagRoutes from './tag/tag.routes';
import typeRoutes from './type/type.routes';
import userRoutes from './user/user.routes';

const router = express.Router();

router.get('/health-check', (req, res) =>
	res.send('OK, it works')
);

router.use('/auth', authRoutes);
router.use('/collections', collectionRoutes);
router.use('/galleries', galleryRoutes);
router.use('/links', linkRoutes);
router.use('/media', mediaRoutes);
router.use('/navigations', navigationRoutes);
router.use('/pages', pageRoutes);
router.use('/posts', postRoutes);
router.use('/roles', roleRoutes);
router.use('/s3', s3Router({
  headers: { 'Access-Control-Allow-Origin': '*' }, // optional
  ACL: 'public-read' // this is default
}));
router.use('/settings', settingRoutes);
router.use('/tags', tagRoutes);
router.use('/types', typeRoutes);
router.use('/users', userRoutes);

export default router;
