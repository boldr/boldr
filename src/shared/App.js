/* @flow */
import React, { Component } from 'react';
import classnames from 'classnames';
import Helmet from 'react-helmet';
import { compose, graphql, gql } from 'react-apollo';
import { connect } from 'react-redux';
import { Route, Switch, Redirect } from 'react-router-dom';
import AsyncComponent from './components/AsyncComponent';
// internal
import { StyleClasses } from './theme/styleClasses';
import Notifications from './components/Notification';
import { getToken } from './core/authentication/token';
// Start routes
import Page from './pages/Page';
import Error404 from './pages/Error404';
import SiteHeaderContainer from './components/SiteHeader/SiteHeaderContainer';
import './styles/main.scss';

const BASE_ELEMENT = StyleClasses.APP;

export const hasAccessToken = () => {
  const token = getToken();
  return !!token;
};

type ProtectedProps = {
  component: ReactElement,
  location: Object,
};
// $FlowIssue
const ProtectedRoute = ({ component: Component, ...rest }: ProtectedProps) =>
  <Route
    {...rest}
    render={props =>
      hasAccessToken
        ? <Component {...props} />
        : <Redirect
            to={{
              pathname: '/login',
              state: { from: props.location },
            }}
          />}
  />;

type Props = {
  className: ?string,
  ui: Object,
  data: Object,
  location: Location,
};

type Location = {
  pathname: string,
  hash: ?string,
};
const AdminDashboard = AsyncComponent({ loader: () => import('./scenes/Admin/AdminDashboard') });

class App extends Component {
  static defaultProps = {
    className: 'app',
  };

  componentDidMount() {
    const jssStyles = document.getElementById('jss-server-side');
    if (jssStyles && jssStyles.parentNode) {
      jssStyles.parentNode.removeChild(jssStyles);
    }
  }
  props: Props;
  render() {
    const { className } = this.props;
    const classes = classnames('boldr', BASE_ELEMENT, className);
    return (
      <div className={classes}>
        <Helmet>
          <html lang="en" />
          <title>Boldr</title>
          <meta name="application-name" content="Boldr" />
          <meta name="description" content="A modern, bold take on a cms" />
          <meta charSet="utf-8" />
          <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          <meta name="theme-color" content="#2b2b2b" />
          <link rel="icon" sizes="16x16 32x32" href="/favicons/favicon.ico" />
          <link
            rel="apple-touch-icon-precomposed"
            sizes="144x144"
            href="/favicons/apple-touch-icon-144x144.png"
          />
          <meta name="msapplication-TileColor" content="#2b2b2b" />
          <meta name="msapplication-TileImage" content="/favicons/mstile-144x144.png" />
          {__DEV__ ? null : <link rel="manifest" href="/manifest.json" />}
        </Helmet>
        <SiteHeaderContainer settings={this.props.data.getSettings} />
        <Switch>
          <ProtectedRoute path="/admin" component={AdminDashboard} />
          <Route path="/" component={Page} />
          <Route component={Error404} />
        </Switch>
        <Notifications />
      </div>
    );
  }
}

export const SETTINGS_QUERY = gql`
  query {
    getSettings {
      id,
      key,
      value,
      label,
      description,
    }
}
`;

const mapStateToProps = state => {
  return {
    router: state.router,
  };
};

export default compose(graphql(SETTINGS_QUERY), connect(mapStateToProps))(App);
