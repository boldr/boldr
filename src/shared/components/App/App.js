/* @flow */
/* eslint-disable global-require */
import React, { Component } from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import type { ReactChildren } from '../../types/react';
import config from '../../core/utils/config';
import Notifications from '../Notification';

if (process.env.NODE_ENV !== 'test') {
  require('../../theme/styles/main.scss');
}

type Props = {
  children: ReactChildren,
};

function App(props: Props) {
  return (
    <div>
      <Helmet
        htmlAttributes={ config('htmlPage.htmlAttributes') }
        titleTemplate={ config('htmlPage.titleTemplate') }
        defaultTitle={ config('htmlPage.defaultTitle') }
        meta={ config('htmlPage.meta') }
        link={ config('htmlPage.links') }
        script={ config('htmlPage.scripts') }
      />
      { React.Children.toArray(props.children) }
      <Notifications />
    </div>
  );
}

export default App;
