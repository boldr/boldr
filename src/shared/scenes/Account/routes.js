// @flow
import React from 'react';
import { Loader, Icon } from 'boldr-ui';
import Loadable from '../../components/Loadable/Loadable';
import SignupContainer from './Signup/SignupContainer';
import LoginContainer from './Login';
import ForgotPassword from './ForgotPassword';
import ResetPassword from './ResetPassword';
import Verify from './Verify';
import PreferencesContainer from './Preferences';

// $FlowIssue
function LoadingComponent({ error }: string) {
  if (error) {
    console.log(error);
    return <p>Error: {error}</p>;
  } else {
    return <Loader />;
  }
}

export default [
  {
    path: '/login',
    exact: true,
    component: LoginContainer,
    routes: [],
  },
  {
    path: '/signup',
    exact: true,
    component: SignupContainer,
    routes: [],
  },
  {
    path: '/account/verify/:token',
    exact: true,
    component: Verify,
    routes: [],
  },
  {
    path: '/account/reset-password/:token',
    exact: true,
    component: ResetPassword,
    routes: [],
  },
  {
    path: '/account/forgot-password',
    exact: true,
    component: ForgotPassword,
    routes: [],
  },
  {
    path: '/account/preferences',
    exact: true,
    component: PreferencesContainer,
    routes: [],
  },
];
