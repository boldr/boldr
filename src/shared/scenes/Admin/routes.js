// @flow
/* eslint-disable */
import React from 'react';
import { Loader, Icon } from 'boldr-ui';
import Loadable from 'react-loadable';
// Admin
import MediaManagerContainer from './Media/MediaManager/MediaManagerContainer';
import UploadMedia from './Media/UploadMedia';
import ArticleEditor from './Content/ArticleEditor';
import NewArticleContainer from './Content/NewArticle/NewArticleContainer';
// import FileManagerContainer
//   from './FileManager/FileManagerContainer';
// import FileEditor from './FileManager/FileEditor';
// import NavigationContainer from './Navigation/NavigationContainer';
import Members from './Members';
// import Settings from './Settings';
import TagsContainer from './Content/Tags/TagsContainer';
import TaggedPost from './Content/Tags/components/TaggedPost/TaggedPost';
//
// function LoadingComponent({ error }) {
//   if (error) {
//     console.log(error);
//     return <p>Error: {error}</p>;
//   } else {
//     return <Loader />;
//   }
// }
const LoadingComponent = (props: { isLoading: boolean, timedOut: boolean, error: boolean }) => {
  if (props.isLoading) {
    // While our other component is loading...
    return props.timedOut ? <div>loader timed out!</div> : <Loader />;
  }
  if (props.error) {
    // If something went wrong
    return <div>Something went wrong...</div>;
  }
  return null;
};
const ArticlesContainer = Loadable({
  // $FlowIssue
  loader: () =>
    import('./Content/Articles/ArticlesContainer' /* webpackChunkName: "dashboard-articles" */),
  loading: props => <LoadingComponent {...props} />
});

const MediaContainer = Loadable({
  loader: () => import('./Media/MediaContainer'),
  loading: props => <LoadingComponent {...props} />
});

export default [
  {
    path: '/admin/content/articles',
    breadcrumb: <Icon kind="shovel" />,
    exact: true,
    component: ArticlesContainer,
    routes: [
      {
        breadcrumb: 'New Article',
        path: '/admin/content/articles/new',
        component: NewArticleContainer,
        routes: [],
      },
      {
        exact: true,
        breadcrumb: 'Edit Article',
        path: '/admin/content/articles/:slug',
        component: ArticleEditor,
        routes: [],
      },
      {
        exact: true,
        breadcrumb: 'Tagged Posts',
        path: '/admin/content/tags/:name',
        component: TaggedPost,
        routes: [],
      },
      {
        exact: true,
        breadcrumb: 'Tags',
        path: '/admin/content/tags',
        component: TagsContainer,
        routes: [],
      },
    ],
  },
  // {
  //       exact: true,
  //       path: '/admin/filemanager',
  //       breadcrumb: <Icon kind="folder-upload" />,
  //       component: FileManagerContainer,
  //       routes: [
  //           {
  //               exact: true,
  //               breadcrumb: 'File Editor',
  //               path: '/admin/filemanager/:id',
  //               component: FileEditor,
  //               routes: [],
  //           },
  //       ],
  //   },
  //   {
  //       exact: true,
  //       path: '/admin/navigation',
  //       breadcrumb: <Icon kind="more" />,
  //       component: NavigationContainer,
  //       routes: [],
  //   },
  // {
  //     exact: true,
  //     path: '/admin/settings',
  //     breadcrumb: <Icon kind="settings" />,
  //     component: Settings,
  //     routes: [],
  // },
  {
    exact: true,
    path: '/admin/members',
    breadcrumb: <Icon kind="account" />,
    component: Members,
    routes: [],
  },
  {
    exact: true,
    path: '/admin/media',
    breadcrumb: <Icon kind="folder-upload" />,
    component: MediaContainer,
    routes: [
      {
        exact: true,
        breadcrumb: 'Upload',
        path: '/admin/media/upload',
        component: UploadMedia,
        routes: [],
      },
      {
        exact: true,
        breadcrumb: 'Manage',
        path: '/admin/media/:id',
        component: MediaManagerContainer,
        routes: [],
      },
    ],
  },
];
