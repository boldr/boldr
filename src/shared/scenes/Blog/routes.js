// @flow
import React from 'react';
import ArticleListing from './ArticleListing';
import Article from './Article/Article';
import TagListContainer from './TagList/TagListContainer';


export default [
  {
    exact: true,
    path: '/blog',
    component: ArticleListing,
    routes: [],
  },
  {
    path: '/blog/:slug',
    exact: true,
    component: Article,
    routes: [],
  },
  {
    exact: true,
    path: '/blog/tags/:name',
    component: TagListContainer,
    routes: [],
  },
];
