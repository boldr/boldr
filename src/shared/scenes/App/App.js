import '../../styles/main.scss';
import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import _ from 'lodash';
import classnames from 'classnames';
import { renderRoutes } from 'react-router-config';

import { StyleClasses } from '../../theme/theme';

import Notifications from '../../components/Notification';

const BASE_ELEMENT = StyleClasses.APP;


import routes from '../../routes';
// Use it when sub routes are added to any route it'll work
export const routeWithSubRoutes = route => (
  <Route
    key={ _.uniqueId() }
    exact={ route.exact || false }
    path={ route.path }
    render={ props => (
      // Pass the sub-routes down to keep nesting
      <route.component { ...props } routes={ route.routes } />
    ) }
  />
);
export default () => {
  return (
    <div>
      <Switch>
        {
          routes.map(route => routeWithSubRoutes(route))
        }
        {renderRoutes(routes)}
      </Switch>
      <Notifications />
    </div>
  );
};
