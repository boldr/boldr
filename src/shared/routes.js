/* eslint-disable */
import React from 'react';
import Loadable from 'react-loadable';
import { Loader } from 'boldr-ui';
import type { Dispatch } from './types/redux';

import App from './components/App';
import Home from './pages/Home';
import About from './pages/About';
import Error404 from './pages/Error404';
// Account
import LoginContainer from './scenes/Account/Login/LoginContainer';
import SignupContainer from './scenes/Account/Signup/SignupContainer';
import ForgotPassword from './scenes/Account/ForgotPassword';
import ResetPassword from './scenes/Account/ResetPassword';
import Verify from './scenes/Account/Verify';
// Profile
import ProfileContainer from './scenes/Profile/ProfileContainer';
// Blog
import PostListingContainer
  from './scenes/Blog/PostListing/PostListingContainer';
import SinglePost from './scenes/Blog/SinglePost/SinglePost';
import TagListContainer from './scenes/Blog/TagList/TagListContainer';
// Admin
import MediaManagerContainer
  from './scenes/Admin/Media/MediaManager/MediaManagerContainer';
import UploadMedia from './scenes/Admin/Media/UploadMedia';
import PostEditor from './scenes/Admin/Post/PostEditor';
import NewPostContainer from './scenes/Admin/Post/NewPost/NewPostContainer';
import FileManagerContainer
  from './scenes/Admin/FileManager/FileManagerContainer';
import FileEditor from './scenes/Admin/FileManager/FileEditor';
import Navigation from './scenes/Admin/Navigation';
import Members from './scenes/Admin/Members';
import Settings from './scenes/Admin/Settings';
import TagsContainer from './scenes/Admin/Tags/TagsContainer';
import TaggedPost from './scenes/Admin/Tags/components/TaggedPost';
import DashboardContainer from './scenes/Admin/Dashboard/DashboardContainer';

// Async data loading actions

import { fetchSettingsIfNeeded } from './state/modules/boldr/settings';
import {
  fetchMembersIfNeeded,
  fetchStatsIfNeeded,
  fetchActivityIfNeeded,
  fetchAttachmentsIfNeeded,
} from './scenes/Admin/state';
import { fetchMediaIfNeeded } from './state/modules/media/actions';
import { fetchProfileIfNeeded } from './state/modules/users';

import { fetchMainMenuIfNeeded } from './state/modules/boldr/menu/actions';
import {
  fetchPostsIfNeeded,
  fetchPostIfNeeded,
  fetchTagsIfNeeded,
  fetchTagPostsIfNeeded,
} from './scenes/Blog/state';

function LoadingComponent({ error }) {
  if (error) {
    console.log(error);
    return <p>Error: {error}</p>;
  } else {
    return <Loader />;
  }
}
const PostListContainer = Loadable({
  loader: () => import('./scenes/Admin/Post/PostList/PostListContainer'),
  LoadingComponent,
});
const AdminDashboard = Loadable({
  loader: () => import('./scenes/Admin/AdminDashboard'),
  LoadingComponent,
});
const MediaContainer = Loadable({
  loader: () => import('./scenes/Admin/Media/MediaContainer'),
  LoadingComponent,
});
export default [
  {
    component: App,
    loadData: async (dispatch: Dispatch) =>
      Promise.all([await dispatch(fetchSettingsIfNeeded())]),
    routes: [
      {
        path: '/',
        exact: true,
        component: Home,
      },
      {
        path: '/about',
        exact: true,
        component: About,
      },
      {
        path: '/blog',
        exact: true,
        component: PostListingContainer,
        loadData: async (dispatch: Dispatch) =>
          Promise.all([
            await dispatch(fetchPostsIfNeeded()),
            await dispatch(fetchTagsIfNeeded()),
          ]),
      },
      {
        path: '/blog/:slug',
        exact: true,
        component: SinglePost,
        loadData: async (dispatch: Dispatch, params: Object) =>
          Promise.all([await dispatch(fetchPostIfNeeded(params.slug))]),
      },
      {
        path: '/blog/tags/:name',
        exact: true,
        component: TagListContainer,
        loadData: async (dispatch: Dispatch, params: Object) =>
          Promise.all([await dispatch(fetchTagPostsIfNeeded(params.name))]),
      },
      {
        path: '/account/login',
        exact: true,
        component: LoginContainer,
      },
      {
        path: '/account/signup',
        exact: true,
        component: SignupContainer,
      },
      {
        path: '/account/forgot-password',
        exact: true,
        component: ForgotPassword,
      },
      {
        path: '/account/reset-password/:token',
        exact: true,
        component: ResetPassword,
      },
      {
        path: '/account/verify/:token',
        exact: true,
        component: Verify,
      },
      {
        path: '/profiles/:username',
        component: ProfileContainer,
        loadData: async (dispatch: Dispatch, params: Object) =>
          Promise.all([await dispatch(fetchProfileIfNeeded(params.username))]),
      },
      {
        path: '/admin',
        component: AdminDashboard,
        auth: {
          required: true,
          redirect: '/account/login',
          status: 307,
        },
        routes: [
          {
            path: '/admin/posts',
            exact: true,
            strict: true,
            component: PostListContainer,
            loadData: async (dispatch: Dispatch) =>
              Promise.all([await dispatch(fetchPostsIfNeeded())]),
          },
          {
            path: '/admin/new-post',
            exact: true,
            component: NewPostContainer,
          },
          {
            path: '/admin/post-editor/:slug',
            exact: true,
            component: PostEditor,
            loadData: async (dispatch: Dispatch, params: Object) =>
              Promise.all([await dispatch(fetchPostIfNeeded(params.slug))]),
          },
          {
            path: '/admin/filemanager',
            exact: true,
            component: FileManagerContainer,
            loadData: async (dispatch: Dispatch) =>
              Promise.all([await dispatch(fetchAttachmentsIfNeeded())]),
          },
          {
            path: '/admin/file-editor/:id',
            exact: true,
            component: FileEditor,
          },
          {
            path: '/admin/navigation',
            exact: true,
            component: Navigation,
            loadData: async (dispatch: Dispatch) =>
              Promise.all([await dispatch(fetchMainMenuIfNeeded())]),
          },
          {
            path: '/admin/members',
            exact: true,
            component: Members,
            loadData: async (dispatch: Dispatch) =>
              Promise.all([await dispatch(fetchMembersIfNeeded())]),
          },
          {
            path: '/admin/settings',
            exact: true,
            component: Settings,
          },
          {
            path: '/admin/tags',
            exact: true,
            component: TagsContainer,
            loadData: async (dispatch: Dispatch) =>
              Promise.all([await dispatch(fetchTagsIfNeeded())]),
          },
          {
            path: '/admin/tags/:name',
            exact: true,
            component: TaggedPost,
          },
          {
            path: '/admin/media',
            exact: true,
            component: MediaContainer,
            loadData: async (dispatch: Dispatch) =>
              Promise.all([await dispatch(fetchMediaIfNeeded())]),
          },
          {
            path: '/admin/media/upload',
            exact: true,
            component: UploadMedia,
          },
          {
            path: '/admin/media/:id',
            exact: true,
            component: MediaManagerContainer,
          },
        ],
      },
      {
        path: '*',
        component: Error404,
      },
    ],
  },
];
