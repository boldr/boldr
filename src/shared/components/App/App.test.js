import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import routes from '../../routes';
import App from './App';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('<App />', () => {
  it('renders snapshot', () => {
    const store = mockStore({});
    const WrapperComp = shallow(
      <Provider store={store}><App route={routes} /></Provider>,
    );
    expect(shallowToJson(WrapperComp)).toMatchSnapshot();
  });
});
