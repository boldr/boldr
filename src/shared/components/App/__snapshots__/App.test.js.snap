// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders snapshot 1`] = `
<Connect(App)
  route={
    Array [
      Object {
        "component": [Function],
        "loadData": [Function],
        "routes": Array [
          Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          Object {
            "component": [Function],
            "exact": true,
            "path": "/about",
          },
          Object {
            "component": [Function],
            "exact": true,
            "loadData": [Function],
            "path": "/blog",
          },
          Object {
            "component": [Function],
            "exact": true,
            "loadData": [Function],
            "path": "/blog/:slug",
          },
          Object {
            "component": [Function],
            "exact": true,
            "loadData": [Function],
            "path": "/blog/tags/:name",
          },
          Object {
            "component": [Function],
            "exact": true,
            "path": "/account/login",
          },
          Object {
            "component": [Function],
            "exact": true,
            "path": "/account/signup",
          },
          Object {
            "component": [Function],
            "exact": true,
            "path": "/account/forgot-password",
          },
          Object {
            "component": [Function],
            "exact": true,
            "path": "/account/reset-password/:token",
          },
          Object {
            "component": [Function],
            "exact": true,
            "path": "/account/verify/:token",
          },
          Object {
            "component": [Function],
            "loadData": [Function],
            "path": "/profiles/:username",
          },
          Object {
            "auth": Object {
              "redirect": "/account/login",
              "required": true,
              "status": 307,
            },
            "component": [Function],
            "path": "/admin",
            "routes": Array [
              Object {
                "component": [Function],
                "exact": true,
                "loadData": [Function],
                "path": "/admin/posts",
                "strict": true,
              },
              Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/new-post",
              },
              Object {
                "component": [Function],
                "exact": true,
                "loadData": [Function],
                "path": "/admin/post-editor/:slug",
              },
              Object {
                "component": [Function],
                "exact": true,
                "loadData": [Function],
                "path": "/admin/filemanager",
              },
              Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/file-editor/:id",
              },
              Object {
                "component": [Function],
                "exact": true,
                "loadData": [Function],
                "path": "/admin/navigation",
              },
              Object {
                "component": [Function],
                "exact": true,
                "loadData": [Function],
                "path": "/admin/members",
              },
              Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/settings",
              },
              Object {
                "component": [Function],
                "exact": true,
                "loadData": [Function],
                "path": "/admin/tags",
              },
              Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/tags/:name",
              },
              Object {
                "component": [Function],
                "exact": true,
                "loadData": [Function],
                "path": "/admin/media",
              },
              Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/media/upload",
              },
              Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/media/:id",
              },
            ],
          },
          Object {
            "component": [Function],
            "path": "*",
          },
        ],
      },
    ]
  }
/>
`;
